channeld/channeld.c

```diff
@@ -2576,7 +2577,7 @@ static void handle_offer_htlc(struct peer *peer, const u8 *inmsg)
 		goto failed;
 	case CHANNEL_ERR_CHANNEL_CAPACITY_EXCEEDED:
 		failcode = WIRE_TEMPORARY_CHANNEL_FAILURE;
-		failmsg = tal_fmt(inmsg, "Capacity exceeded");
+		failmsg = tal_fmt(inmsg, "Capacity exceeded by %s", fmt_amount_msat(inmsg, &exceed));
```

Comment:
This will only be propagated to a local `sendpay` failure, incidentally.  If the failing HTLC is from a forward, we specifically do not propagate `failmsg` back to the payer, as we do not want to leak information to the payer.

Reply:
Xref.

https://github.com/ElementsProject/lightning/blob/11412435ce0cae92205178db9a9f4b64d0d78830/lightningd/peer_htlcs.c#L360-L389

Notice how `failurestr` is only used in the path where `hout->am_origin` (i.e."I am the origin" of this outgoing HTLC, i.e. a local `pay`/`sendpay` attempt).  In the path where there is an incoming HTLC attached to the outgoing HTLC (i.e. a forward) `failurestr` is specifically not used.

Reply:
@ZmnSCPxj 
Having the exception just locally actually the intention behind.

The current channel amounts are, from a p2p protocol perspective, not know by intention.
However, hiding the exceeded amounts make it hard for local stuff (like plugins) to use a channel in its full volume (See ugly loop to determine correct HTLC fee in `drain` https://github.com/lightningd/plugins/pull/22/files#diff-04c04e1a674a88f0db43dd254913a601R332 )

If we expose the exceeded amounts (that depend on dynamic stuff like onchain fees) such plugin do only need to catch once to have the correct value and not loop in order to make an approximation.